src/workflows-app/utils/submission-utils.js

```diff
@@ -162,7 +162,8 @@ export const typeMatch = (cbasType, wdsType) => {
     );
   }
   if (unwrappedCbasType.type === 'array') {
-    return _.startsWith('ARRAY_OF_')(wdsType) && typeMatch(unwrappedCbasType.array_type, _.replace('ARRAY_OF_', '')(wdsType));
+    // should match single values as well as array values
+    return typeMatch(unwrappedCbasType.array_type, _.replace('ARRAY_OF_', '')(wdsType));
```

Comment:
The only gotcha I can think of here - which I'm not sure whether WDS even supports - is that WDS could be supplying an array that's **more** nested than the one CBAS wants. 

eg 
| WDS Type  | CBAS Type   | Valid  |
|---|---|---|
| ARRAY_OF_INT  | Array[Int]  | ✅   | 
| INT  | Array[Int]  | ✅   | 
| ARRAY_OF_ARRAY_OF_INT  | Array[Int]  | ❌ | 

Does the `_.replace('ARRAY_OF_', '')(wdsType)` remove all of the possible ARRAY_OF prefixes or just the outer one?

And out of interest.... do these WDS types also support optionals? Because that's another potential "wrapper" type that probably interacts weirdly with our coercions

Reply:
I don't think WDS supports nested arrays:
https://github.com/DataBiosphere/terra-workspace-data-service/blob/b0817c4b8f7f8a7ad084944b2a04077cd55bf453/service/src/main/java/org/databiosphere/workspacedataservice/service/model/DataTypeMapping.java#L8-L25

As for optionals, I don't think WDS is concerned with them either. If CBAS fails to convert these examples in the future (or WDS expands supported data types) we'll still get an error back post-submission, and I think in this case we'd rather let users run into those than not be able to submit at all as we saw in this case

Reply:
TIL that WDS has explicit types for all of those arrays. Ok, not a problem then 👍